apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx

```diff
@@ -126,22 +126,24 @@ export default function DocumentEditor({
       providerNames: providers.map((p) => p.name) ?? [],
       integrationNames: integrations?.map((i) => i.name) ?? [],
     })
-  }, [document.promptlVersion, agentToolsMap, providers, integrations])
+  }, [
+    document.promptlVersion,
+    agentToolsMap,
+    providers,
+    integrations,
+    document,
+    documents,
+    runReadMetadata,
+    value,
+  ])
```

Comment:
hmm check that the documents array is not triggering more effects than necessary